File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/blockmonitor/BlockMonitorFragment.java
package com.didichuxing.doraemonkit.kit.blockmonitor;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.View;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.kit.blockmonitor.core.BlockMonitorManager;
import com.didichuxing.doraemonkit.kit.core.BaseFragment;
import com.didichuxing.doraemonkit.kit.core.SettingItem;
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter;
import com.didichuxing.doraemonkit.widget.titlebar.HomeTitleBar;

/**
 * @desc: 卡顿检测首页
 */

public class BlockMonitorFragment extends BaseFragment {
    private static final String TAG = "BlockMonitorIndexFragment";
    public static final String KEY_JUMP_TO_LIST = "KEY_JUMP_TO_LIST";

    @Override
    protected int onRequestLayout() {
        return R.layout.dk_fragment_block_monitor_index;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
    }


    private void initView() {
        HomeTitleBar title = findViewById(R.id.title_bar);
        title.setListener(new HomeTitleBar.OnTitleBarClickListener() {
            @Override
            public void onRightClick() {
                getActivity().finish();
            }
        });
        RecyclerView mSettingList = findViewById(R.id.setting_list);
        mSettingList.setLayoutManager(new LinearLayoutManager(getContext()));
        SettingItemAdapter settingItemAdapter = new SettingItemAdapter(getContext());
        mSettingList.setAdapter(settingItemAdapter);
        settingItemAdapter.append(new SettingItem(R.string.dk_item_block_switch, BlockMonitorManager.getInstance().isRunning()));
        settingItemAdapter.append(new SettingItem(R.string.dk_item_block_goto_list));
        settingItemAdapter.append(new SettingItem(R.string.dk_item_block_mock));

        settingItemAdapter.setOnSettingItemSwitchListener(new SettingItemAdapter.OnSettingItemSwitchListener() {
            @Override
            public void onSettingItemSwitch(View view, SettingItem data, boolean on) {
                if (data.desc == R.string.dk_item_block_switch) {
                    if (on) {
                        BlockMonitorManager.getInstance().start();
                    } else {
                        BlockMonitorManager.getInstance().stop();
                    }
                }
            }
        });
        settingItemAdapter.setOnSettingItemClickListener(new SettingItemAdapter.OnSettingItemClickListener() {
            @Override
            public void onSettingItemClick(View view, SettingItem data) {
                if (data.desc == R.string.dk_item_block_goto_list) {
                    showContent(BlockListFragment.class);
                } else if (data.desc == R.string.dk_item_block_mock) {
                    mockBlock();
                }
            }
        });

        if (getArguments() != null) {
            boolean jump = getArguments().getBoolean(KEY_JUMP_TO_LIST, false);
            if (jump) {
                showContent(BlockListFragment.class);
            }
        }
    }

    private void mockBlock() {
        try {
            getView().postDelayed(new Runnable() {
                @Override
                public void run() {
                    try {
                        Thread.sleep(2000);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            }, 1000);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/blockmonitor/BlockMonitorFragment.kt
package com.didichuxing.doraemonkit.kit.blockmonitor

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.blockmonitor.BlockListFragment
import com.didichuxing.doraemonkit.kit.blockmonitor.core.BlockMonitorManager.Companion.instance
import com.didichuxing.doraemonkit.kit.core.BaseFragment
import com.didichuxing.doraemonkit.kit.core.SettingItem
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter.OnSettingItemClickListener
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter.OnSettingItemSwitchListener
import com.didichuxing.doraemonkit.widget.titlebar.HomeTitleBar
import kotlinx.android.synthetic.main.dk_fragment_block_monitor_index.*

/**
 * @desc: 卡顿检测首页
 */
class BlockMonitorFragment : BaseFragment() {
    companion object {
        private const val TAG = "BlockMonitorIndexFragment"
        const val KEY_JUMP_TO_LIST = "KEY_JUMP_TO_LIST"
    }

    override fun onRequestLayout(): Int {
        return R.layout.dk_fragment_block_monitor_index
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
    }

    private fun initView() {
        title_bar.setListener(object : HomeTitleBar.OnTitleBarClickListener {
            override fun onRightClick() {
                activity?.finish()
            }
        })
        val mSettingList = findViewById<RecyclerView>(R.id.setting_list)
        mSettingList.layoutManager = LinearLayoutManager(context)
        val settingItemAdapter = SettingItemAdapter(context)
        mSettingList.adapter = settingItemAdapter
        settingItemAdapter.append(SettingItem(R.string.dk_item_block_switch, instance.isRunning))
        settingItemAdapter.append(SettingItem(R.string.dk_item_block_goto_list))
        settingItemAdapter.append(SettingItem(R.string.dk_item_block_mock))
        settingItemAdapter.setOnSettingItemSwitchListener(object : OnSettingItemSwitchListener {
            override fun onSettingItemSwitch(view: View, data: SettingItem, on: Boolean) {
                if (data.desc == R.string.dk_item_block_switch) {
                    if (on) {
                        instance.start()
                    } else {
                        instance.stop()
                    }
                }
            }
        })
        settingItemAdapter.setOnSettingItemClickListener(object : OnSettingItemClickListener {
            override fun onSettingItemClick(view: View, data: SettingItem) {
                if (data.desc == R.string.dk_item_block_goto_list) {
                    showContent(BlockListFragment::class.java)
                } else if (data.desc == R.string.dk_item_block_mock) {
                    mockBlock()
                }
            }
        })
        if (arguments != null) {
            val jump = arguments!!.getBoolean(KEY_JUMP_TO_LIST, false)
            if (jump) {
                showContent(BlockListFragment::class.java)
            }
        }
    }

    private fun mockBlock() {
        try {
            view?.postDelayed({
                try {
                    Thread.sleep(2000)
                } catch (e: InterruptedException) {
                    //ignore
                }
            }, 1000)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        super.onDestroy()
    }


}